### ISSUE TITLE
How to evaluate the validation set every n steps?

### ISSUE THREAD

**Issue by huntzhan** [allenai/allennlp#2786]
Please first search our GitHub repository for similar questions.  If you don't find a similar example you can use the following template:

**System (please complete the following information):**

irrelevant

**Question**

I'd like to find an easy way to trigger validation every n steps. For example, I have a training set of 10k batches and I want to evaluate on the validation set every 100 batches.

**Comment by joelgrus** [comment]
There's not an easy way to do this right now, other than subclassing our `Trainer` and making your custom subclass do what you want. We are working on redesigning the trainer to use events and callbacks, after which it would be relatively simple to accomplish this, but that doesn't help you today.

Since the validation runs at the end of each epoch, one workaround would be to artificially size each "epoch" to represent 100 batches, which you can by specifying `instances_per_epoch` in your data iterator:

https://github.com/allenai/allennlp/blob/master/allennlp/data/iterators/data_iterator.py#L39
